Common/src/Pollster/PostProcessor.cs

```diff
@@ -29,12 +29,23 @@
 
 namespace ArmoniK.Core.Common.Pollster;
 
+/// <summary>
+///   A background service that handles post-processing of completed tasks.
+///   It reads task handlers from the post-processing queue and finalizes them
+///   by performing any necessary cleanup operations and recording the success.
```

Comment:
It is not only cleanup. It is also result uploading, task submissions, and message acknowledgment.